### DISCUSSION TITLE
Transcoded videos

### DISCUSSION THREAD

**Discussion by jjspierx** [immich-app/immich#5685] [General]
Is there a list of video formats out there that will prevent my videos from being transcoded?  I am currently using an iPhone 12, and saving videos as "High Efficiency", which stores the files with a .mov extension.  However, I think the video format is HEVC @60fps.  When these files are uploaded into Immich, they are then transcoded, and stored in the encoded-videos directory.  This means the original file is still being stored in the libary folder (it is 159mb) and the transcoded video is stored in the encoded-videos folder (and is 221mb).  I assume the reason for this is to store a version of the video that will play in all of the Immich clients.   However, this means that the asset is taking up more than double the size of the original file.  For users with a lot of video assets, this could get out of hand quickly.  I have a relatively small library compared to many of you out there and I already have a 20gb encoded-video directory.  

1. If there is a list of acceptable formats that will not be transcoded, that would be great.

2.  Is there a list of which clients (Android, iOS, Web) can play which formats?  If so, it would be great if I could choose whether or not to transcode videos.  For instance, if my videos are being transcoded just for compatability with Immich for Android, it would be nice if I could tell Immich not to transcode, so I will not be using an Android client.

3.  This is probably a long-shot, and I may be shot for even suggesting a "destructive" option, but what about the option "delete original" after transcoding?  Google Photos has a section called "Utilities" that allows the user to auto-archive screenshots and other stuff like that.  How about a section for "converting transcoded videos" or something like that allows the user to "convert" a transcoded video into the library video?  If the user chose this option, the original would be deleted, the transcoded video would be moved from the encoded-video directory into the library, and the db asset would be updated to reflect the new location.  It would be nice to rename to the encode video to match the original name (except the extension of course).

**Comment by zackpollard** [comment]
1.) You can play with the transcoding configuration options, if you don't want encoded videos you can turn that feature off entirely, but you may lose compatibility on certain clients. You can also configure in what cases the files will be transcoded there, although the options are limited.
2.) [caniuse.com](https://caniuse.com/?search=hevc) can show you for example the compatibility of hevc in different browsers, any modern iOS or android devices will support hevc and h264 natively.
3.) We are currently not looking at implementing this kind of feature. Immich is at its heart a backup application. Our view is that storage is cheap in the grand scheme of things and we want to preserve the original files in all cases.

**Reply by snimavat** [reply]
+1 for "Immich is at its heart a backup application" - thts is what intended use is.

**Comment by jjspierx** [comment]
Thanks for the info.  And totally understand on the viewpoint of #3.  Even I was conflicted about it. :-)

**Comment by Batwam** [comment]
I am having the same issue as well importing my Photo/Videos library with the encoded-video folder taking way too much space. Unfortunately, storage on my micro-pc is at a premium. In my case, I have a 150GB collection of photos on a 500GB hard drive and the local immich folder is 80GB already (process still in progress)...

**Reply by jjspierx** [reply]
Click on "Administration" on the top navbar, right side.  Then click settings from the Left tab-bar.  Scroll down till you see "Video Transcoding Settings".  Then scroll down to "Transcode Policy".  Change setting to "Don't transcode any videos, may break playback on some clients".

**Reply by mio-19** [reply]
After changing this settings, How do I remove encoded-video safely?

**Reply by Snickers333** [reply]
I believe you can do it by running a "Transcode Videos" job for all files. It applies the current policy for all existing videos.

**Comment by meminens** [comment]
Does anyone know how to force encode all videos? I changed encoding settings from H264 to VP9. I then went to administration settings and then job. Clicked on the ALL button in the encoding box. But it the total number of videos quickly goes down to 0 in less than a minute. I don't think Immich force re-encodes my videos to the new option.

**Reply by Snickers333** [reply]
You must have changed something in the policy so that these videos are skipped

**Comment by WhyNotHugo** [comment]
Given that video transcoding is only required for playback "on some clients", could it be disabled by default, and enabled if any such client ever connects?

This feature takes up a huge amount of CPU and tens of gigabytes of space. There's no Linux client, so I'm never going to use any external client.
